tests/components/sharkiq/test_shark_iq.py

```diff
@@ -0,0 +1,283 @@
+    """Test all of the shark vacuum property accessors."""
+    ayla_api = MockAyla()
+    shark_vac = _get_mock_shark_vac(ayla_api)
+    coordinator = SharkIqUpdateCoordinator(hass, None, ayla_api, [shark_vac])
+    shark = SharkVacuumEntity(shark_vac, coordinator)
+
+    assert shark.name == "Sharknado"
+    assert shark.serial_number == "AC000Wxxxxxxxxx"
+    assert shark.model == "RV1000A"
+
+    assert shark.battery_level == 50
+    assert shark.fan_speed == "Eco"
+    shark.sharkiq.set_property_value(Properties.POWER_MODE, 0)
+    assert shark.fan_speed == "Normal"
+    assert isinstance(shark.recharge_resume, bool) and shark.recharge_resume
+    assert isinstance(shark.low_light, bool) and not shark.low_light
+
+    target_state_attributes = {
+        ATTR_ERROR_CODE: 7,
+        ATTR_ERROR_MSG: "Cliff sensor is blocked",
+        ATTR_RECHARGE_RESUME: True,
+        ATTR_LOW_LIGHT: False,
+    }
+    state_json = json.dumps(shark.device_state_attributes, sort_keys=True)
+    target_json = json.dumps(target_state_attributes, sort_keys=True)
+    assert state_json == target_json
+
+    assert not shark.should_poll
+
+
+@patch.object(SharkIqVacuum, "set_property_value", new=_set_property)
+@patch.object(SharkIqVacuum, "async_set_property_value", new=_async_set_property)
+async def test_shark_metadata(hass: HomeAssistant) -> None:
+    """Test shark properties coming from metadata."""
+    ayla_api = MockAyla()
+    shark_vac = _get_mock_shark_vac(ayla_api)
+    coordinator = SharkIqUpdateCoordinator(hass, None, ayla_api, [shark_vac])
+    shark = SharkVacuumEntity(shark_vac, coordinator)
+    shark.sharkiq._update_metadata(  # pylint: disable=protected-access
+        SHARK_METADATA_DICT
+    )
+
+    target_device_info = {
+        "identifiers": {("sharkiq", "AC000Wxxxxxxxxx")},
+        "name": "Sharknado",
+        "manufacturer": "Shark",
+        "model": "RV1001AE",
+        "sw_version": "Dummy Firmware 1.0",
+    }
+    state_json = json.dumps(shark.device_info, sort_keys=True)
+    target_json = json.dumps(target_device_info, sort_keys=True)
+    assert state_json == target_json
+
+
+def _get_async_update(err=None):
+    async def _async_update(_) -> bool:
+        if err is not None:
+            raise err
+        return True
+
+    return _async_update
+
+
+@patch.object(AylaApi, "async_list_devices", new=_async_list_devices)
+async def test_updates(hass: HomeAssistant) -> None:
+    """Test the update coordinator update functions."""
+    ayla_api = get_ayla_api(TEST_USERNAME, TEST_PASSWORD)
+    shark_vac = _get_mock_shark_vac(ayla_api)
+    mock_config = MagicMock(spec=ConfigEntry)
+    coordinator = SharkIqUpdateCoordinator(hass, mock_config, ayla_api, [shark_vac])
+
+    with patch.object(SharkIqVacuum, "async_update", new=_get_async_update()):
+        update_called = (
+            await coordinator._async_update_data()  # pylint: disable=protected-access
+        )
+    assert update_called
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6090606)